src/python/pants/option/option_types.py

```diff
@@ -14,6 +15,11 @@
 _ValueT = TypeVar("_ValueT")
 # NB: We don't provide constraints, as our `XListOption` types act like a set of contraints
 _ListMemberType = TypeVar("_ListMemberType")
+_HelpT = Union[Callable[[Any], str], str]
+
+
+def _eval_maybe_helpt(help: _HelpT, subsystem_cls: Any) -> str:
+    return help(subsystem_cls) if inspect.isfunction(help) else help  # type: ignore
```

Comment:
Prefer to use the MyPy error code: https://www.pantsbuild.org/v2.10/docs/style-guide#use-error-codes-in--type-ignore-comments

Reply:
Remind me at some point to codify these conventions in a flake8 plugin ;)

Reply:
https://github.com/pantsbuild/flake8-pantsbuild :D

Reply:
Those codes are underwhelming. It needs to be beefed up, and in-repo

Reply:
Moving this in-repo would be great! That wasn't possible two years ago. But now that we have support for multiple resolves + in-source Flake8 plugins (thanksssss)